forge/forge/tensor.py

```diff
@@ -223,6 +226,12 @@ def __repr__(self):
             return f"Forge Tensor: {self.value()}, {self.data_format}"
         return f"Forge Empty Tensor: {self.shape}"
 
+    def ndim(self) -> int:
```

Comment:
Should we just name this function `rank()`? 

E.g. `... Got { predictions.rank() } ...`

Reply:
torch uses ndim, so i've used it also; i will rename it to rank

Reply:
Actually jax also uses ndim.
As I think about it, 'rank' is kind of an overloaded notation - if you ask for a rank of a 2d tensor, do you want to get the matrix rank or nr of dimensions?
I would stick with ndim. Thoughts? @pilkicTT @nvukobratTT

Reply:
I am fine either way 😄 

I also thought that `ndim` seems more aligned with rest of the world...

Reply:
Leaving as `ndim`...